src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -512,7 +534,22 @@ private async Task<(bool, string)> GetManifestPackageUpdate(SdkFeatureBand sdkFe
             }
         }
 
-        private string GetAdvertisingManifestPath(SdkFeatureBand featureBand, ManifestId manifestId) =>
+        private async Task<(bool, NuGetVersion)> GetPackageVersion(PackageId packageId, PackageSourceLocation packageSourceLocation = null, bool includePreview = false)
+        {
+            try
+            {
+                var latestVersion = await _nugetPackageDownloader.GetLatestPackageVersion(packageId, packageSourceLocation: _packageSourceLocation, includePreview: includePreview);
+                return (true, latestVersion);
+            }
+            catch
```

Comment:
I think we don't want to catch all exceptions here, just `NuGetPackageNotFoundException`